libs/feature/dataviz/src/lib/chart-view/chart-view.component.ts

```diff
@@ -129,10 +154,34 @@ export class ChartViewComponent {
       }
     })
   )
-  chartType: InputChartType = 'bar'
-  xProperty$ = new BehaviorSubject<string>(undefined)
-  yProperty$ = new BehaviorSubject<string>(undefined)
-  aggregation$ = new BehaviorSubject<FieldAggregation[0]>('sum')
+
+  get permalinkBaseUrl() {
+    return this.permalinkBaseUrlToken?.charAt(0) === '/'
+      ? `${window.location.origin}${this.permalinkBaseUrlToken}`
+      : this.permalinkBaseUrlToken
+  }
+
+  permalink$ = combineLatest([
+    this.xProperty$.pipe(filter((value) => value !== undefined)),
+    this.yProperty$.pipe(filter((value) => value !== undefined)),
+    this.aggregation$,
+    this.chartType$,
+  ]).pipe(
+    map(([xProp, yProp, aggregation, chartType]) => {
+      const link = `${this.permalinkBaseUrl}
+?e=gn-dataset-view-chart
+&a=api-url=${this.config.basePath}
+&a=aggregation=${aggregation}
+&a=x-axis=${xProp}
```

Comment:
I haven't follow the devs, but one little remark here. I think we have to be consistent in the naming for all our API
- internal API (code)
- iframe API (params)
- web component API (attributes)

Why `xProp` then `x-axis` ?

Reply:
Good reminder, here it was just the local variable in the stream, which I adapted. The exposed component attributes (internal API) are `xAxis`...

Reply:
Ok, I renamed `xAxis` to `xProperty` everywhere to be consistent here.